internal/controllers/cluster/cluster_controller_status_test.go

```diff
@@ -0,0 +1,2336 @@
+				Reason:  clusterv1.ClusterDeletingWaitingForBeforeClusterDeleteHookV1Beta2Reason,
+				Message: "Deleting: Some message",
+			},
+		},
+		{
+			name: "Drops messages from InfraCluster and ControlPlane when using fallback to fields; drop message from RemoteConnectionProbe when true",
+			cluster: &clusterv1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "machine-test",
+					Namespace: metav1.NamespaceDefault,
+				},
+				Spec: clusterv1.ClusterSpec{
+					InfrastructureRef: &corev1.ObjectReference{Kind: "AWSCluster"},
+					ControlPlaneRef:   &corev1.ObjectReference{Kind: "KubeadmControlPlane"},
+				},
+				Status: clusterv1.ClusterStatus{
+					InfrastructureReady: true,
+					ControlPlaneReady:   true,
+					V1Beta2: &clusterv1.ClusterV1Beta2Status{
+						Conditions: []metav1.Condition{
+							{
+								Type:    clusterv1.ClusterInfrastructureReadyV1Beta2Condition,
+								Status:  metav1.ConditionTrue,
+								Reason:  "Foo",
+								Message: infrastructureReadyFallBackMessage("AWSCluster", true),
+							},
+							{
+								Type:    clusterv1.ClusterControlPlaneAvailableV1Beta2Condition,
+								Status:  metav1.ConditionTrue,
+								Reason:  "Foo",
+								Message: controlPlaneAvailableFallBackMessage("KubeadmControlPlane", true),
+							},
+							{
+								Type:   clusterv1.ClusterWorkersAvailableV1Beta2Condition,
+								Status: metav1.ConditionTrue,
+								Reason: "Foo",
+							},
+							{
+								Type:    clusterv1.ClusterRemoteConnectionProbeV1Beta2Condition,
+								Status:  metav1.ConditionTrue,
+								Reason:  clusterv1.ClusterRemoteConnectionProbeSucceededV1Beta2Reason,
+								Message: remoteConnectionProbeSucceededMessage,
+							},
+							{
+								Type:   clusterv1.ClusterDeletingV1Beta2Condition,
+								Status: metav1.ConditionFalse,
+								Reason: "Foo",
+							},
+							{
+								Type:    "MyAvailabilityGate",
+								Status:  metav1.ConditionFalse,
+								Reason:  "SomeReason",
+								Message: "Some message",
+							},
+						},
+					},
+				},
+			},
+			expectCondition: metav1.Condition{
+				Type:   clusterv1.ClusterAvailableV1Beta2Condition,
+				Status: metav1.ConditionTrue,
+				Reason: v1beta2conditions.MultipleInfoReportedReason,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			setAvailableCondition(ctx, tc.cluster)
+
+			condition := v1beta2conditions.Get(tc.cluster, clusterv1.ClusterAvailableV1Beta2Condition)
+			g.Expect(condition).ToNot(BeNil())
+			g.Expect(*condition).To(v1beta2conditions.MatchCondition(tc.expectCondition, v1beta2conditions.IgnoreLastTransitionTime(true)))
+		})
+	}
+}
+
+type fakeClusterOption interface {
+	ApplyToCluster(c *clusterv1.Cluster)
+}
+
+func fakeCluster(name string, options ...fakeClusterOption) *clusterv1.Cluster {
+	c := &clusterv1.Cluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: clusterv1.GroupVersion.String(),
+			Kind:       "Cluster",
```

Comment:
Do we depend on this being set? I don't know if it is actually set in the controller

Reply:
The code already depends on it, because it is used by all the ownerRef checks.

Reply:
Are we setting it somewhere explicitly? I didn't find it. The Cluster coming from the cached client shouldn't have it set IIRC

Reply:
Existing code relies on controller runtime setting this field, which unfortunately is gray area.
If we want to reduce this reliance, let's follow up with a PR fixing this across the board. The downside of it will be that this will hide regression on the controller runtime side...